File: leetcode/3-Palindrome-Number/palindrome-number.java
//Runtime: 8 ms, faster than 92.81% of Java online submissions for Palindrome Number.
//Memory Usage: 34.7 MB, less than 100.00% of Java online submissions for Palindrome Number.
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        ArrayList<Integer> inter = new ArrayList();
        while (x > 0) {
            inter.add(x % 10);
            x = x / 10;
        }
        for (int i = 0; i < inter.size() / 2; i++) {
            if (inter.get(i) != inter.get(inter.size() - i - 1)) {
                return false;
            }
        }
        return true;
    }
}

File: leetcode/3-Palindrome-Number/palindrome-number.cpp
//Runtime: 60 ms, faster than 82.57% of C++ online submissions for Palindrome Number.
//Memory Usage: 17.5 MB, less than 97.56% of C++ online submissions for Palindrome Number.
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0){
            return false;
        }
        deque<int> inter;
        while(x > 0){
            inter.push_back(x % 10);
            x = x / 10;
        }
        while(! inter.empty()){
            if (inter.front() != inter.back()){
                return false;
            }
            inter.pop_back();
            if (! inter.empty()){
                inter.pop_front();
            }
        }
        return true;
    }
};